### PR TITLE
Improved the ergonomics of our Bazel rules.

### PR DESCRIPTION
**Pull request by alxrsngrtn**
This commit breaks away the rules portion of #6039.

Here, the build_defs/internal *.bzl files have been reorganized. Macros are now streamlined and belong to appropriately named files.

CC: @bayesianmind @csilvestrini

### CODE REVIEW

File: third_party/java/arcs/build_defs/internal/plan.bzl [hunk 1]
```diff
@@ -1,37 +1,134 @@
...
+        srcs = outs,
+        platforms = platforms,
+        visibility = visibility,
+        deps = arcs_sdk_deps + deps,
     )
+    return {"outs": outs, "deps": arcs_sdk_deps}
 
-    return [DefaultInfo(files = depset([output]))]
-
-recipe2plan = rule(
```
Comment (bgogul) [inline third_party/java/arcs/build_defs/internal/plan.bzl:20]: Removing this from here will cause problems importing the PR. Do you want to leave this here for now in addition to `rules.bzl`? That way you can import the PR first, make change to tools.google3.bzl internally and then subsequently delete it?

File: third_party/java/arcs/build_defs/internal/gen.bzl [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      visibility: visibility of the generated arcs_kt_library
+    """
+
+    manifest_name = name + "_manifest"
+    schema_name = name + "_schema"
+    plan_name = name + "_plan"
+
+    arcs_manifest(
+        name = manifest_name,
+        srcs = manifest_only(srcs),
```
[the remark this answers is not in the document]
Comment (alxrsngrtn) [reply third_party/java/arcs/build_defs/internal/gen.bzl:43]: This was intentional. And -- I think I want to revert this change. I like that `arcs_manifest` checks to see that all srcs end with `.arcs`.

File: third_party/java/arcs/build_defs/internal/gen.bzl [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        manifest_proto = False,
+        deps = deps,
```
[the remark this answers is not in the document]
Comment (alxrsngrtn) [reply third_party/java/arcs/build_defs/internal/gen.bzl:45]: I intended this change, but am having second thoughts. I will add it back.

File: third_party/java/arcs/build_defs/internal/plan.bzl (hunk 1 above)
Comment (alxrsngrtn) [reply third_party/java/arcs/build_defs/internal/plan.bzl:20]: I like that plan of attack.

File: third_party/java/arcs/build_defs/internal/schemas.bzl [hunk 4]
```diff
@@ -3,26 +3,18 @@
...
-        name,
-        src,
-        deps,
-        out,
-        language_name,
-        language_flag,
-        wasm,
-        test_harness = False):
-    """Generates source code for the given .arcs schema file.
+def arcs_cc_schema(name, src, deps = [], out = None):
```
Comment (csilvestrini) [inline third_party/java/arcs/build_defs/internal/schemas.bzl:9]: It looks like you moved arcs_kt_shcema to kotlin.bzl. Do you want to move arcs_cc_schema to a file called cpp.bzl or cc.bzl too?

**Review by csilvestrini — APPROVED** [review]

File: third_party/java/arcs/build_defs/internal/schemas.bzl (hunk 4 above)
Comment (alxrsngrtn) [reply third_party/java/arcs/build_defs/internal/schemas.bzl]: I like that; updated.

**Pull request merged**